src/engine/InputQuery.cpp

```diff
@@ -239,6 +238,31 @@ InputQuery &InputQuery::operator=( const InputQuery &other )
         }
     }
 
+    // Setting plConstraints and topological order
+    if ( !other._networkLevelReasoner )
+    {
+        for ( const auto &constraint : other._plConstraints )
+            _plConstraints.append( constraint->duplicateConstraint() );
+    }
+    else
+    {
+        for ( const auto &constraint : other._networkLevelReasoner->
+                  getConstraintsInTopologicalOrder() )
+        {
+            if ( !other._plConstraints.exists( constraint ) )
```

Comment:
How can this happen? The constraint appears in the "TopologicalOrder" list, but not in the plc list?

Reply:
This used to happen because the topological order is constructed before this line https://github.com/NeuralNetworkVerification/Marabou/blob/cb7bb66c4958068f56aaf1c3061ee2817846d168/src/engine/Preprocessor.cpp#L712

Now I remove a constraint from the topological order when it is deleted from the inputQuery.